src/styles/shared/_controls.scss

```diff
@@ -11,87 +11,155 @@
+      border-color: transparent;
 
       &::after {
-        top: -4px;
-        right: -4px;
-        bottom: -4px;
-        left: -4px;
-        background: none;
-        border-radius: border-radius() + 1px;
+        background: color('white');
       }
-    }
-  } @else if $style == focused {
-    background: color('indigo');
-    box-shadow: 0 0 0 1px color('indigo'), shadow(transparent);
 
-    &::after {
-      background: color('white');
-    }
+      @media (-ms-high-contrast: active) {
+        &::after {
+          border: 1px dashed ms-high-contrast-color('button-text');
+        }
+      }
+    } @else if $style == error {
+      background: color('red', 'dark');
+      box-shadow: 0 0 0 1px transparent;
 
-    @media (-ms-high-contrast: active) {
       &::after {
-        border: 1px dashed ms-high-contrast-color('button-text');
+        background: color('red', 'lighter');
       }
-    }
-  } @else if $style == error {
-    background: color('red', 'dark');
-    box-shadow: 0 0 0 1px transparent;
+    } @else if $style == disabled {
+      background: color('sky');
+      box-shadow: none;
 
-    &::after {
-      background: color('red', 'lighter');
-    }
-  } @else if $style == disabled {
-    background: color('sky');
-    box-shadow: none;
+      &::after {
+        background: color('sky', 'lighter');
+      }
 
-    &::after {
-      background: color('sky', 'lighter');
-    }
+      @media (-ms-high-contrast: active) {
+        color: ms-high-contrast-color('disabled-text');
+      }
+    } @else if $style == focused-error {
+      background: color('indigo');
+      box-shadow: 0 0 0 1px color('indigo');
 
-    @media (-ms-high-contrast: active) {
-      color: ms-high-contrast-color('disabled-text');
+      &::after {
+        background: color('red', 'lighter');
+      }
     }
-  } @else if $style == focused-error {
-    background: color('indigo');
-    box-shadow: 0 0 0 1px color('indigo');
+    // design-language
+  } @else if $global-theming == true {
+    --control-border-width: #{rem(2px)};
+
+    @if $style == base {
+      position: relative;
+      border: var(--control-border-width) solid var(--p-border-secondary);
+      background-color: var(--p-surface);
+      border-radius: var(--p-border-radius-base);
+      will-change: border-color;
+      transition: border-color var(--p-duration-1-0-0) var(--p-ease-in);
+
+      &::before {
+        content: '';
+        position: absolute;
+        top: calc(var(--control-border-width) * -1);
+        right: calc(var(--control-border-width) * -1);
+        bottom: calc(var(--control-border-width) * -1);
+        left: calc(var(--control-border-width) * -1);
+        border-radius: var(--p-border-radius-base);
+        background-color: var(--p-action-interactive);
+        opacity: 0;
+        transform: scale(0.25);
+        transition: opacity var(--p-duration-1-0-0) linear,
+          transform var(--p-duration-1-0-0) linear;
```

Comment:
Two things about these two lines. 

1. It looks like the prototype is using 150ms for check mark transition
1. It's also using a bezier curve for the function `cubic-bezier(0.4, 0.22, 0.28, 1)`